### PR TITLE
bump k8s to 1.29 and kind to 1.28

### PR DESCRIPTION
**Pull request by ishankhare07**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement
/kind feature
/kind test

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves ENG-2571

**Please provide a short message that should be published in the vcluster release notes**
Fixed an issue where vcluster ...

**What else do we need to know?**

### CODE REVIEW

File: pkg/values/k8s.go [hunk 1]
```diff
@@ -7,27 +7,31 @@ import (
...
+	"1.27": "registry.k8s.io/kube-scheduler:v1.27.8",
+	"1.26": "registry.k8s.io/kube-scheduler:v1.26.11",
+	"1.25": "registry.k8s.io/kube-scheduler:v1.25.16",
 }
 
 var K8SEtcdVersionMap = map[string]string{
+	"1.29": "registry.k8s.io/etcd:3.5.10-0",
 	"1.28": "registry.k8s.io/etcd:3.5.9-0",
 	"1.27": "registry.k8s.io/etcd:3.5.7-0",
 	"1.26": "registry.k8s.io/etcd:3.5.6-0",
```
Comment (FabianKramm) [inline pkg/values/k8s.go:37]: we also add the new version below where we still check for 28

File: .github/workflows/e2e.yaml [hunk 2]
```diff
@@ -161,7 +161,7 @@ jobs:
         uses: engineerd/setup-kind@v0.5.0
         with:
           version: "v0.20.0"
-          image: kindest/node:v1.27.3
+          image: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
```
Comment (FabianKramm) [inline .github/workflows/e2e.yaml:164]: why use a sha here?

**Review by FabianKramm — CHANGES_REQUESTED** [review]

File: pkg/values/k8s.go (hunk 1 above)
Comment (ishankhare07) [reply pkg/values/k8s.go:37]: for the kind [releases page](https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0)

> Additional images built for this release:
    1.28: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
    

>   NOTE: You must use the @sha256 digest to guarantee an image built for this release, until such a time as we switch to a different tagging scheme. Even then we will highly encourage digest pinning for security and reproducibility reasons.

shall I give it a try without the sha once?

File: .github/workflows/e2e.yaml (hunk 2 above)
Comment (ishankhare07) [reply .github/workflows/e2e.yaml:164]: from the kind [releases page](https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0)

> Additional images built for this release:
    1.28: kindest/node:v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31
    

>   NOTE: You must use the @sha256 digest to guarantee an image built for this release, until such a time as we switch to a different tagging scheme. Even then we will highly encourage digest pinning for security and reproducibility reasons.

shall I give it a try without the sha once?

**Review by FabianKramm — APPROVED** [review]

**Pull request merged**